src/store/selectors/activities/index.ts

```diff
@@ -0,0 +1,456 @@
+import { Big } from "big.js";
+
+import {
```

Comment:
Hmm, so this looks fine to me logic-wise. However, it does strike me that we're essentially running the same reduction that we perform on operations to generate the members state, including pretty much the same validation to check the validity of each operation before applying. And, this selector recalculates activities every time it gets called.

The trickiness seems to be that we're relying on members state to generate these activities. However, we only rely on members state as generated up-to the current operation we're creating the activity for.

Perhaps the "member" reducer could be come "memoized", and "applyOperation" could first apply the operation against memoized members state and then against memoized activity state.

Reply:
We can memoize selectors automatically with `reselect` so that it doens't have to be calculated every time, I'll save that for a future PR.

Secondarily the reduction is the same right now, because it's 1:1 between operations and activities, but very soon that will not be the case—it will be radically different once we start combining operations and stuff. The validity checks are much laxer here—just that once the member state has all been put together, that the resulting activites are actually valid given that state.

So I don't really agree with this comment, but I think that `reselect` would solve the performance concerns (as well as offloading this to the API, which is the ultimate endpoint).

Reply:
`reselect` sgtm.

-- Disclaimer -- both of the below concerns are probs not that big of a deal in light of us both doing serious validation of operations in the API itself and maintaining activities and member state in firebase instead of reconstructing it locally each time. That being said.. if you care to read the following:
---

Why should the validity checks be looser here? I don't see why we'd want to include operations that don't validate as we construct member state in constructing our activity state. In fact - I can construct a case right now where an op would validate against member state but not actually be valid -- any op that happens prior to the createMember operation, or, alternately, more sophisticated checks like "does the member have enough raha to perform this give operation?"

I also think that the reducer is being applied over the memoized state - so it's okay if operations and activities don't have a 1:1 relationship - the you can create 0, 1, many activities on the memoized state for each operation.